src/component/mxgraph/shape/IconPainter.ts

```diff
@@ -59,13 +62,40 @@ export default class IconPainter {
     const margin = StyleUtils.getMargin(shape.style);
 
     return {
-      c,
+      canvas,
       ratioFromParent,
-      shape: { x, y, w: width, h: height, strokeWidth: shapeStrokeWidth },
+      shape: { x, y, width, height, strokeWidth: shapeStrokeWidth },
       icon: { isFilled, fillColor, strokeColor, strokeWidth: iconStrokeWidth, margin },
     };
   }
 
+  private static translateIconToShapeCenter(c: mxgraph.mxXmlCanvas2D, shape: ShapeConfiguration, iconWidth: number, iconHeight: number): void {
```

Comment:
ℹ️ As the envelop & circle icons are not paint with the same way that the others, we need to have another way to resize the icon & scale it.

We refactor this during the refatoring PR of the scale